android/src/main/java/com/oblador/keychain/KeychainModule.java

```diff
@@ -13,291 +20,586 @@
+                                              String minimumSecurityLevel,
+                                              ReadableMap unusedOptions,
+                                              @NonNull final Promise promise) {
+    setGenericPasswordForOptions(server, username, password, minimumSecurityLevel, promise);
+  }
+
+  @ReactMethod
+  public void getInternetCredentialsForServer(@NonNull String server,
+                                              ReadableMap unusedOptions,
+                                              @NonNull final Promise promise) {
+    getGenericPasswordForOptions(server, promise);
+  }
+
+  @ReactMethod
+  public void resetInternetCredentialsForServer(@NonNull String server,
+                                                ReadableMap unusedOptions,
+                                                @NonNull final Promise promise) {
+    resetGenericPasswordForOptions(server, promise);
+  }
+
+  @ReactMethod
+  public void getSupportedBiometryType(@NonNull final Promise promise) {
+    try {
+      boolean fingerprintAuthAvailable = isFingerprintAuthAvailable();
+
+      if (fingerprintAuthAvailable) {
+        promise.resolve(FINGERPRINT_SUPPORTED_NAME);
+      } else {
+        promise.resolve(null);
+      }
+    } catch (Exception e) {
+      Log.e(KEYCHAIN_MODULE, e.getMessage(), e);
+
+      promise.reject(Errors.E_SUPPORTED_BIOMETRY_ERROR, e);
+    } catch (Throwable fail) {
+      Log.e(KEYCHAIN_MODULE, fail.getMessage(), fail);
+
+      promise.reject(Errors.E_UNKNOWN_ERROR, fail);
+    }
+  }
+  //endregion
+
+  //region Implementation
+
+  /** Is provided access control string matching biometry use request? */
+  public static boolean getUseBiometry(@Nullable final String accessControl) {
+    return accessControl != null
+      && (accessControl.equals(ACCESS_CONTROL_BIOMETRY_ANY)
+      || accessControl.equals(ACCESS_CONTROL_BIOMETRY_CURRENT_SET));
+  }
+
+  private void addCipherStorageToMap(@NonNull final CipherStorage cipherStorage) {
+    cipherStorageMap.put(cipherStorage.getCipherStorageName(), cipherStorage);
+  }
+
+  /**
+   * Extract credentials from current storage. In case if current storage is not matching
+   * results set then executed migration.
+   */
+  @NonNull
+  private DecryptionResult decryptCredentials(@NonNull final String alias,
+                                              @NonNull final CipherStorage current,
+                                              @NonNull final ResultSet resultSet)
+    throws CryptoFailedException, KeyStoreAccessException {
+    final String storageName = resultSet.cipherStorageName;
+
+    // The encrypted data is encrypted using the current CipherStorage, so we just decrypt and return
+    if (storageName.equals(current.getCipherStorageName())) {
+      final DecryptionResultHandler handler = getInteractiveHandler(current);
+      current.decrypt(handler, alias, resultSet.username, resultSet.password, ANY);
```

Comment:
To follow code style used in this library:
```suggestion
      current.decrypt(handler, alias, resultSet.username, resultSet.password, SecurityLevel.ANY);
```